http/oidc/src/main/java/org/wildfly/security/http/oidc/ServerRequest.java

```diff
@@ -241,4 +245,39 @@ public String getError() {
         }
     }
 
+    public static AccessAndIDTokenResponse getBearerToken(OidcClientConfiguration oidcClientConfiguration, String username, String password) throws Exception {
+        AccessAndIDTokenResponse tokenResponse;
+        HttpClient client = oidcClientConfiguration.getClient();
+
+        HttpPost post = new HttpPost(oidcClientConfiguration.getTokenUrl());
+        List<NameValuePair> formparams = new ArrayList<>();
+        formparams.add(new BasicNameValuePair(GRANT_TYPE, PASSWORD));
+        formparams.add(new BasicNameValuePair(USERNAME, username));
+        formparams.add(new BasicNameValuePair(PASSWORD, password));
+
+        ClientCredentialsProviderUtils.setClientCredentials(oidcClientConfiguration, post, formparams);
+        UrlEncodedFormEntity form = new UrlEncodedFormEntity(formparams, "UTF-8");
+        post.setEntity(form);
+
+        HttpResponse response = client.execute(post);
+        int status = response.getStatusLine().getStatusCode();
+        HttpEntity entity = response.getEntity();
+        if (status != HttpStatus.SC_OK) {
+            EntityUtils.consumeQuietly(entity);
+            throw log.unableToObtainToken(status);
+        }
+        if (entity == null) {
+            throw log.noMessageEntity();
+        }
+        InputStream is = entity.getContent();
+        try {
```

Comment:
Just a minor, this could be chaged to try with resources:

try (InputStream is = entity.getContent()) {
            tokenResponse = JsonSerialization.readValue(is, AccessAndIDTokenResponse.class);
        }